server/methods/setStatusMessage.coffee

```diff
@@ -0,0 +1,12 @@
+Meteor.methods
+	setStatusMessage: (status, message) ->
+		userId = Meteor.userId()
+		if not userId
+			throw new Meteor.Error('invalid-user', "[methods] toogleFavorite -> Invalid user")
+
+		console.log '[methods] setStatusMessage -> '.green, 'userId:', Meteor.userId(), 'arguments:', arguments
+
+		user = Meteor.user()
+		setField = { statusMessages:user.profile.statusMessages}
+		setField.statusMessages[status] = message
+		result = Meteor.users.update userId, {$set: { "profile": setField}}
```

Comment:
We're overwriting the other "profile" fields (e.g location, access, etc).  try setting the profile.statusMessages field directly.  i.e.
```javascript
statusMessages = user?.profile?.statusMessages || {}
statusMessages[status] = message
Meteor.users.update userId, {$set : {"profile.statusMessages" : statusMessages}}
```